cmd/otel-allocator/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/otel-allocator
 
-go 1.17
+go 1.18
```

Comment:
Isn't this the minimum version that is required to compile? Iam asking [because of](https://github.com/open-telemetry/opentelemetry-operator/pull/784#issuecomment-1070528328):
> at RH there is probably no go 1.18 image for productization. We have to make sure the project can be built with go 1.17 as well

Reply:
I am able to build it with go 1.17 without issues (e.g. by changing go builder version in the dockerfile)

Reply:
oh ok, but it would allow people to use 1.18 features, right? Would then make sense to keep it at 1.17 until we have a build image?

Reply:
That is a valid argument. If somebody will use 1.18 features we will have to fork :/

Reply:
1.18 features make MUCH sense for Kubernetes Operators, like the support for generics. Much of the code can be simplified with that, I presume. But we are not in a hurry to change to it, though.